modules/sor/lib/poolsV2/gyro2/gyro2Pool.ts

```diff
@@ -56,7 +47,7 @@ export class Gyro2Pool implements BasePool {
             const tokenAmount = TokenAmount.fromScale18Amount(token, scale18);
 
             poolTokens.push(
-                new Gyro2PoolToken(token, tokenAmount.amount, poolToken.index, parseEther(poolToken.priceRate)),
```

Comment:
Just a double check why its no longer Gyro2PoolToken?

Reply:
Previous implementation of pool tokens with rate was done separately for each pool type, but that added repeated code and it was actually missing important stuff (e.g. update scale18 amounts on increase/decrease operations).
A while ago I pushed a PR replacing all of those with `PoolTokenWithRate`, but apparently I missed this one.
I took the opportunity to fix it.